src/commands/readme.rs

```diff
@@ -231,24 +231,98 @@ fn find_readme_template<'a>(
     })
 }
 
-fn fix_doc_links(pkg_name: &str, readme: &str, doc_uri: Option<&str>) -> String {
+fn rewrite_doc_links(pkg_name: &str, readme: &str, doc_uri: Option<&str>) -> String {
     RELATIVE_LINKS_REGEX
-        .replace_all(&readme, |caps: &Captures| match caps.name("url") {
-            Some(url) if url.as_str().starts_with("../") => format!(
-                "[{}]({}{})",
-                &caps.name("text").unwrap().as_str(),
-                doc_uri.unwrap_or(DEFAULT_DOC_URI),
-                &url.as_str().replace('_', "-").replace("/index.html", "")[3..]
-            ),
-            Some(url) if url.as_str().starts_with("./") => format!(
-                "[{}]({}{}/latest/{}/{})",
-                &caps.name("text").unwrap().as_str(),
-                doc_uri.unwrap_or(DEFAULT_DOC_URI),
-                pkg_name,
-                pkg_name.replace('-', "_"),
-                &url.as_str()[2..]
-            ),
-            _ => caps[0].to_string(),
+        .replace_all(&readme, |caps: &Captures| {
+            rewrite_matched_doc_link(caps, pkg_name, doc_uri)
         })
         .into()
 }
+
+fn rewrite_matched_doc_link(caps: &Captures, pkg_name: &str, doc_uri: Option<&str>) -> String {
+    match caps.name("url") {
+        // Skip absolute links
+        Some(url) if url.as_str().starts_with("http") => caps[0].to_string(),
+        // Handle relative links to sibling crate
+        Some(url) if url.as_str().starts_with("../") => make_sibling_doc_link(
+            caps.name("text").unwrap().as_str(),
+            &url.as_str()[3..],
+            doc_uri,
+        ),
+        // Handle relative links to current crate
+        Some(url) => make_relative_doc_link(
+            caps.name("text").unwrap().as_str(),
+            if url.as_str().starts_with("./") {
```

Comment:
I feel like we could use `str::strip_prefix` more often here but that's nitpicking